src/scanner.cc

```diff
@@ -75,14 +77,87 @@ struct Scanner {
     }
   }
 
-  // Scans the multistring start. Assumes that the m indicating a multistring
-  // was already consumed.
+  bool valid_symtag_start(int32_t lookahead) {
+    return ('a' <= lookahead && lookahead <= 'z') || ('A' <= lookahead && lookahead <= 'Z');
+  }
+
+  bool valid_symtag_middle(int32_t lookahead) {
+    return valid_symtag_start(lookahead) || ('0' <= lookahead && lookahead <= '9') || lookahead == '-' || lookahead == '\'' || lookahead == '_';
+  }
+
+  bool scan_until_sstr_start_end(TSLexer *lexer, bool m_scanned) {
+    enum RecognizerState {
```

Comment:
Same, having a few comment lines to explain why we need that for could be more posterity-proof.